pyiceberg/transforms.py

```diff
@@ -521,6 +556,16 @@ def project(self, name: str, pred: BoundPredicate[L]) -> Optional[UnboundPredica
         else:
             raise ValueError(f"Could not project: {pred}")
 
+    def strict_project(self, name: str, pred: BoundPredicate[Any]) -> Optional[UnboundPredicate[Any]]:
+        if isinstance(pred, BoundUnaryPredicate):
+            return pred.as_unbound(Reference(name))
+        elif isinstance(pred, BoundLiteralPredicate):
+            return pred.as_unbound(Reference(name), pred.literal)
+        elif isinstance(pred, BoundSetPredicate):
+            return pred.as_unbound(Reference(name), pred.literals)
+        else:
+            raise ValueError(f"Could not project: {pred}")
```

Comment:
I am a little bit confused here: why do we only raise error in IdentityTransform?

Reply:
Good question. It is code that's not being hit since we can always project identity transforms. I've changed it to return None.